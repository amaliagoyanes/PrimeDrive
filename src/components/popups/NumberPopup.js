import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Text,
    View,
    Alert
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {Row, Col} from 'react-native-easy-grid';
import styles from '../../style/styles';
import Modal from 'react-native-root-modal';
import {NumericKeyboard} from '../index.js'
import {scale, scaleText} from '../../scaling'


export default class NumberPopup extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: '', open: false, px: 0, py: 0
        };

        this.numOnPress = this.numOnPress.bind(this);
        this.cleanInput = this.cleanInput.bind(this);
        this.confirm = this.confirm.bind(this);
        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
    }

    open(px, py, uid, negative=false) {
        this.setState({open: true, px: px, py: py, uid: uid, negative: negative})
    }

    close() {
        this.setState({open: false, input: ''})
    }

    numOnPress(text) {
        if (this.props.lockZero && this.state.input.length <= 0 && Number(text) === 0) 
            return;
        let newVal = this.state.input + text,
            max = this.props.max;
        if (max && parseFloat(newVal) > max) {
            this.props.alert(this.props.intl.messages["app.notGreaterThan"] + max);
            return;
        }
        this.setState({input: newVal})
    }

    cleanInput(){
        if (this.props.deleteAll)
            this.setState({input: ''});
        else
            this.setState({input: this.state.input.slice(0, this.state.input.length - 1)});
    }

    confirm(){
        if (this.props.onOk !== undefined)
            this.props.onOk(this.state.negative ? parseFloat(-this.state.input) : parseFloat(this.state.input), null, this.state.uid);
        this.close();
    }

    render() {
        let left = this.props.arrow === 'left' ?
            this.state.px + 10 : this.state.px - Dimensions.get('screen').width * 0.15 - 10;
        let leftArrow = this.props.arrow === 'left' ?
            this.state.px : this.state.px - 15;
        let py = this.state.py > Dimensions.get('screen').height / 2 ? 215 : 50;

        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={this.close}>
                    <View style={{
                        position: 'absolute',
                        top: this.state.py + 15,
                        left: leftArrow,
                    }}>
                        <View
                            style={this.props.arrow === 'left' ? [styles.triangle, styles.leftTriangle] : styles.triangle}/>
                    </View>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: this.state.py - py,
                        left: left,
                        width: Dimensions.get('screen').width * 0.15,
                        height: Dimensions.get('screen').height * 0.4,
                        backgroundColor: '#404048'
                    }}>
                        <NumericKeyboard size={6} numOnPress={this.numOnPress} input={this.state.input}
                                         lockZero={this.props.lockZero}
                                         cleanInput={this.cleanInput} smallBlock intl={this.props.intl}
                                         showDelimiter={this.props.showDelimiter} postfix={this.props.postfix}/>
                        <Row size={1}>
                            <TouchableOpacity style={[styles.tabButton, {width: '100%', backgroundColor: this.state.input.length ? '#2eab61' : '#93D8B1' }]} // '#93D8B1'
                                              onPress={this.confirm} disabled={!this.state.input.length}>
                                <Text style={[styles.text, {fontSize: scaleText(9)}]}>
                                    {this.props.intl !== undefined && this.props.intl.messages["app.okUpper"]}</Text>
                            </TouchableOpacity>
                        </Row>
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}
