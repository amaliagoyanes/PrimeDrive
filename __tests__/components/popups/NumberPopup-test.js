'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import NumberPopup from '../../../src/components/popups/NumberPopup';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = shallow(
        <NumberPopup />);
    tree.setState({open: true})
    expect(tree).toMatchSnapshot();
    expect(tree.find('NumericKeyboard').length).toEqual(1)

});

it('renders correctly with left arrow', () => {
    const tree = shallow(
        <NumberPopup arrow="left"/>);
    tree.setState({open: true});
    expect(tree).toMatchSnapshot();
});

it('renders correctly with right arrow', () => {
    const tree = shallow(
        <NumberPopup arrow="right"/>);
    tree.setState({open: true});
    expect(tree).toMatchSnapshot();
});
