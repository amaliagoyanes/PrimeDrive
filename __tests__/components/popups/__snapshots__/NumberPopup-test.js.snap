// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  onRequestClose={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.1)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "left": -15,
          "position": "absolute",
          "top": 15,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomWidth": 10,
            "borderLeftColor": "#404048",
            "borderLeftWidth": 10,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopWidth": 10,
            "height": 0,
            "width": 0,
          }
        }
      />
    </View>
    <TouchableOpacity
      activeOpacity={1}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#404048",
          "height": 533.6,
          "left": -122.5,
          "position": "absolute",
          "top": -50,
          "width": 112.5,
        }
      }
    >
      <NumericKeyboard
        cleanInput={[Function]}
        input="1"
        numOnPress={[Function]}
        size={6}
        smallBlock={true}
      />
      <RowNB
        size={1}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={false}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#404048",
                "justifyContent": "center",
                "width": "34%",
              },
              Object {
                "backgroundColor": "#2eab61",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Gotham-Book",
                  "fontSize": 14,
                },
                Object {
                  "fontSize": 9,
                },
              ]
            }
          >
            OK
          </Text>
        </TouchableOpacity>
      </RowNB>
    </TouchableOpacity>
  </TouchableOpacity>
</Modal>
`;

exports[`renders correctly with left arrow 1`] = `
<Modal
  onRequestClose={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.1)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": 15,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 10,
              "borderLeftColor": "#404048",
              "borderLeftWidth": 10,
              "borderRightColor": "transparent",
              "borderRightWidth": 0,
              "borderStyle": "solid",
              "borderTopColor": "transparent",
              "borderTopWidth": 10,
              "height": 0,
              "width": 0,
            },
            Object {
              "borderLeftColor": "transparent",
              "borderLeftWidth": 0,
              "borderRightColor": "#404048",
              "borderRightWidth": 10,
            },
          ]
        }
      />
    </View>
    <TouchableOpacity
      activeOpacity={1}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#404048",
          "height": 533.6,
          "left": 10,
          "position": "absolute",
          "top": -50,
          "width": 112.5,
        }
      }
    >
      <NumericKeyboard
        cleanInput={[Function]}
        input="1"
        numOnPress={[Function]}
        size={6}
        smallBlock={true}
      />
      <RowNB
        size={1}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={false}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#404048",
                "justifyContent": "center",
                "width": "34%",
              },
              Object {
                "backgroundColor": "#2eab61",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Gotham-Book",
                  "fontSize": 14,
                },
                Object {
                  "fontSize": 9,
                },
              ]
            }
          >
            OK
          </Text>
        </TouchableOpacity>
      </RowNB>
    </TouchableOpacity>
  </TouchableOpacity>
</Modal>
`;

exports[`renders correctly with right arrow 1`] = `
<Modal
  onRequestClose={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.1)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "left": -15,
          "position": "absolute",
          "top": 15,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomWidth": 10,
            "borderLeftColor": "#404048",
            "borderLeftWidth": 10,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopWidth": 10,
            "height": 0,
            "width": 0,
          }
        }
      />
    </View>
    <TouchableOpacity
      activeOpacity={1}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#404048",
          "height": 533.6,
          "left": -122.5,
          "position": "absolute",
          "top": -50,
          "width": 112.5,
        }
      }
    >
      <NumericKeyboard
        cleanInput={[Function]}
        input="1"
        numOnPress={[Function]}
        size={6}
        smallBlock={true}
      />
      <RowNB
        size={1}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={false}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#404048",
                "justifyContent": "center",
                "width": "34%",
              },
              Object {
                "backgroundColor": "#2eab61",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Gotham-Book",
                  "fontSize": 14,
                },
                Object {
                  "fontSize": 9,
                },
              ]
            }
          >
            OK
          </Text>
        </TouchableOpacity>
      </RowNB>
    </TouchableOpacity>
  </TouchableOpacity>
</Modal>
`;
